verl/trainer/ppo/ray_trainer.py

```diff
@@ -990,6 +1003,19 @@ def fit(self):
                     # generate a batch
                     with _timer("gen", timing_raw):
                         if not self.async_rollout_mode:
+                            workers = self.actor_rollout_wg.workers
+                            if not self._actor_rollout_wg_initialized:
+                                init_process_group(workers)
+                                self._actor_rollout_wg_initialized = True
+
+                            # Prepare state dict for all workers, and get local shards from all workers
+                            # and pass them to rank 0
+                            workers[0].prepare_state_dict.remote()
+                            shards_refs = [worker.prepare_state_dict.options(tensor_transport="NCCL").remote() for worker in workers[1:]]
+                            params = workers[0].gather_params.remote(*shards_refs)
+
+                            workers[0].broadcast_params_and_sync_weights.remote()
+                            ray.get([worker.broadcast_params_and_sync_weights.remote(params) for worker in workers[1:]])
```

Comment:
```suggestion

                            shards_refs = [worker.prepare_state_dict.options(tensor_transport="NCCL").remote() for worker in workers[1:]]
                            params = workers[0].gather_params.remote(*shards_refs)


                            ray.get([worker.broadcast_params_and_sync_weights.remote(params) for worker in workers])
```

Is it possible to get the code to look something like this?

Reply:
done